sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/HttpTimeoutPolicyControlPlaneHotPath.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.http;
+
+import com.azure.cosmos.implementation.Configs;
+import com.azure.cosmos.implementation.RxDocumentServiceResponse;
+import io.netty.handler.codec.http.HttpMethod;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.azure.cosmos.implementation.HttpConstants.StatusCodes.REQUEST_TIMEOUT;
+
+public class HttpTimeoutPolicyControlPlaneHotPath extends HttpTimeoutPolicy {
+
+    final public static HttpTimeoutPolicy instance = new HttpTimeoutPolicyControlPlaneHotPath();
+
+    public HttpTimeoutPolicyControlPlaneHotPath() {
+    }
+
+    @Override
+    public Duration maximumRetryTimeLimit() {
+        return Duration.ofSeconds(Configs.getHttpResponseTimeoutInSeconds());
+    }
+
+    @Override
+    public Integer totalRetryCount() {
+        return getTimeoutAndDelays().size();
+    }
+
+    @Override
+    public List<ResponseTimeoutAndDelays> getTimeoutList() {
+        return getTimeoutAndDelays();
+    }
+
+    @Override
+    public Boolean isSafeToRetry(HttpMethod httpMethod) {
+        return true;
+    }
+
+    // The hot path should always be safe to retires since it should be retrieving meta data
+    // information that is not idempotent.
+    @Override
+    public Boolean shouldRetryBasedOnResponse(HttpMethod requestHttpMethod, Mono<RxDocumentServiceResponse> responseMessage) {
+        if (responseMessage == null) {
+            return false;
+        }
+
+        final AtomicInteger statusCode = new AtomicInteger();
+        responseMessage.flatMap(rm -> {
```

Comment:
This flatMap has no affect, because the return value is ignored. Please fix this.

Reply:
The method is removed, as it was not being used.